### PR TITLE
Rewrite "save and restore data" section

### PR DESCRIPTION
**Pull request by thaJeztah**
This started as a "fix some links", but reading the section, I found various issues that could use some improvement. After that, I noticed that both the Windows and Mac pages had the same content, so I moved this to a common page instead.

Note that there's still additional information that should likely be included (and perhaps a rewrite into "per topic" headings (containers, images, ...).

Things that are missing;

- backup/restore swarmkit configs
- backup/restore swarmkit secrets
- backup/restore CLI configuration, trust data etc.

What I did in this pull request:

- recommend pushing images to Docker Hub as the default approach to back up   images (but kept `docker image save` as an alternative)
- recommend `docker container commit` to save containers to an image, as doing  so preserves both the container's configuration _and_ filesystem changes.  `docker container export` only preserves the filesystem, but all other   configuration is lost, which makes it hard to restore a container from it. Using `docker container commit` (committing the container to an image) also allows to follow the same procedure to backup containers as for images (both allowing them to be pushed to a registry, or to a file).
- added some information about "wether or not" containers should be backed up, as it may not be needed to back up containers if all data is stored  in volumes (best practice).
- added some warnings regarding "sensitive data" (environment variables), and links to making repositories private before pushing.

### CODE REVIEW

**Comment by thaJeztah** [comment]
@usha-mandya @stephen-turner ptal

**Review by stephen-turner — COMMENTED** [review]
It looks plausible to me, but I'm not the best person to review the technical details. But I trust Sebastiaan.

**Comment by thaJeztah** [comment]
@StefanScherer @chris-crone perhaps one of you could have a quick peek if it lgty as well?

File: _data/toc.yaml [hunk 1]
```diff
@@ -1258,6 +1258,8 @@ manuals:
     title: Dashboard
   - path: /desktop/faqs/
     title: FAQs
+  - path: /desktop/backup-and-restore/
+    title: Backup and restore data
```
Comment (usha-mandya) [inline _data/toc.yaml:1262]:
```suggestion
    title: Back up and restore data
```

File: desktop/backup-and-restore.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+title: Backup and restore data
```
Comment (usha-mandya) [inline desktop/backup-and-restore.md:2]:
```suggestion
title: Back up and restore data
```

File: docker-for-mac/install.md [hunk 3]
```diff
@@ -111,4 +93,6 @@ For information on how to back up and restore data volumes, see [Backup, restore
   to run and submit diagnostics, and submit issues.
 - [FAQs](../desktop/faqs.md) provide answers to frequently asked questions.
 - [Release notes](release-notes.md) lists component updates, new features, and improvements associated with Docker Desktop releases.
-- [Get started with Docker](/get-started/) provides a general Docker tutorial.
+- [Get started with Docker](../get-started/index.md) provides a general Docker tutorial.
+* [Backup and restore data](../desktop/backup-and-restore.md) provides instructions
```
Comment (usha-mandya) [inline docker-for-mac/install.md:97]:
```suggestion
* [Back up and restore data](../desktop/backup-and-restore.md) provides instructions
```

File: docker-for-windows/install.md [hunk 4]
```diff
@@ -120,32 +120,13 @@ To uninstall Docker Desktop from your Windows machine:
 2. Select **Docker Desktop** from the **Apps & features** list and then select **Uninstall**.
 3. Click **Uninstall** to confirm your selection.
 
-> **Note** 
+> **Important**
 >
-> Uninstalling Docker Desktop will destroy Docker containers and images local to the machine and remove the files generated by the application.
+> Uninstalling Docker Desktop destroys Docker containers, images, volumes, and
+> other Docker related data local to the machine, and removes the files generated
+> by the application. Refer to the [backup and restore data](../desktop/backup-and-restore.md)
```
Comment (usha-mandya) [inline docker-for-windows/install.md:127]:
```suggestion
> by the application. Refer to the [back up and restore data](../desktop/backup-and-restore.md)
```

File: docker-for-windows/install.md [hunk 5]
```diff
@@ -156,3 +137,5 @@ For information on how to back up and restore data volumes, see [Backup, restore
   how to get support.
 * [FAQs](../desktop/faqs.md) provides answers to frequently asked questions.
 * [Release notes](release-notes.md) lists component updates, new features, and improvements associated with Docker Desktop releases.
+* [Backup and restore data](../desktop/backup-and-restore.md) provides instructions
```
Comment (usha-mandya) [inline docker-for-windows/install.md:140]:
```suggestion
* [Back up and restore data](../desktop/backup-and-restore.md) provides instructions
```

File: desktop/backup-and-restore.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+   > Should I back up my containers?
+   >
+   > If you use volumes or bind-mounts to store your container data, backing up
+   > your containers may not be needed, but make sure to remember the options that
+   > were used when creating the container or use a [Docker Compose file](../compose/compose-file/index.md)
+   > if you want to re-create your containers with the same configuration after
+   > re-installation.
+
+2. Use [`docker pusn`](../engine/reference/commandline/push.md) to push any
```
Comment (usha-mandya) [inline desktop/backup-and-restore.md:37]:
```suggestion
2. Use [`docker push`](../engine/reference/commandline/push.md) to push any
```

**Review by StefanScherer — COMMENTED** [review]
@thaJeztah Yes, the rewrite makes sense. With Usha's change requests it will LGTM. :-)

**Comment by thaJeztah** [comment]
oh, thanks for the ping; looks like I missed Usha's comments when she left them; let me update

**Review by usha-mandya — APPROVED** [review]
LGTM. Thanks @thaJeztah

**Pull request merged**